src/utils/contract.test.ts

```diff
@@ -0,0 +1,168 @@
+import {getStatesFromSourceCode} from "./contract";
+
+describe('getStatesfromSourceCode', () => {
+  test('elementary types', async () => {
+    const sourceCode = `
+    contract Test {
+      string public constant name = "Uniswap";
+
+      uint8 public constant decimals = 18;
+
+      int256 public immutable integers = -21;
+
+      bool internal constant boolean = true;
+
+      address private constant plainAddress = '0x1ca7c995f8eF0A2989BbcE08D5B7Efe50A584aa1'
+
+      address payable constant payableAddress = '0xbb38B6F181541a6cEdfDac0b94175B2431Aa1A02'
+
+      bytes32 constant byteText = "ByteText";
+    }
+    `;
+
+    const states = await getStatesFromSourceCode(sourceCode);
+    expect(states).toStrictEqual([
+      { slot: 0, type: 'string name;', variable: 'name' },
+      { slot: 1, type: 'uint8 decimals;', variable: 'decimals' },
```

Comment:
actually `constant is not stored in evm storage. correct slot mapping will be:
 name - slot 0
 integers -  slot 1
 plainAddress - slot 2
payableAddress - slot 3
byteText - slot 4

Reply:
https://medium.com/aigang-network/how-to-read-ethereum-contract-storage-44252c8af925

> Constants are not stored in a storage. Available only in code.

Reply:
Constants have been removed from those declarations and new tests added.